frontend/projects/ui/src/app/services/patch-db/patch-db.service.ts

```diff
@@ -1,165 +1,49 @@
+import { Observable, of, Subscription } from 'rxjs'
 import {
   catchError,
   debounceTime,
   filter,
   finalize,
-  mergeMap,
-  shareReplay,
   switchMap,
   take,
   tap,
-  withLatestFrom,
 } from 'rxjs/operators'
-import { pauseFor } from '@start9labs/shared'
+import { ConnectionService, PatchConnection } from '../connection.service'
 import { DataModel } from './data-model'
-import { ApiService } from '../api/embassy-api.service'
-import { AuthService } from '../auth.service'
-import { BOOTSTRAPPER, PATCH_SOURCE, PATCH_SOURCE$ } from './patch-db.factory'
-
-export enum PatchConnection {
-  Initializing = 'initializing',
-  Connected = 'connected',
-  Disconnected = 'disconnected',
-}
+import { BOOTSTRAPPER } from './patch-db.factory'
 
 @Injectable({
   providedIn: 'root',
 })
 export class PatchDbService {
-  private readonly WS_SUCCESS = 'wsSuccess'
-  private readonly patchConnection$ = new ReplaySubject<PatchConnection>(1)
-  private readonly wsSuccess$ = new BehaviorSubject(false)
-  private readonly polling$ = new BehaviorSubject(false)
-  private subs: Subscription[] = []
-
-  readonly connected$ = this.watchPatchConnection$().pipe(
-    filter(status => status === PatchConnection.Connected),
-    take(1),
-    shareReplay(),
-  )
+  private sub?: Subscription
 
   constructor(
-    // [wsSources, pollSources]
-    @Inject(PATCH_SOURCE) private readonly sources: Source<DataModel>[],
     @Inject(BOOTSTRAPPER)
     private readonly bootstrapper: Bootstrapper<DataModel>,
-    @Inject(PATCH_SOURCE$)
-    private readonly sources$: BehaviorSubject<Source<DataModel>[]>,
-    private readonly http: ApiService,
-    private readonly auth: AuthService,
-    private readonly storage: Storage,
     private readonly patchDb: PatchDB<DataModel>,
+    private readonly connectionService: ConnectionService,
   ) {}
 
-  init() {
-    this.sources$.next([this.sources[0], this.http])
-    this.patchConnection$.next(PatchConnection.Initializing)
-  }
-
   async start(): Promise<void> {
-    this.init()
-
-    this.subs.push(
-      // Connection Error
-      this.patchDb.connectionError$
-        .pipe(
-          debounceTime(420),
-          withLatestFrom(this.polling$),
-          mergeMap(async ([e, polling]) => {
-            if (polling) {
-              console.log('patchDB: POLLING FAILED', e)
-              this.patchConnection$.next(PatchConnection.Disconnected)
-              await pauseFor(2000)
-              this.sources$.next([this.sources[1], this.http])
-              return
-            }
-
-            console.log('patchDB: WEBSOCKET FAILED', e)
-            this.polling$.next(true)
-            this.sources$.next([this.sources[1], this.http])
-          }),
-        )
-        .subscribe({
-          complete: () => {
-            console.warn('patchDB: SYNC COMPLETE')
-          },
+    this.sub = this.patchDb.cache$
+      .pipe(
+        debounceTime(420),
+        tap(cache => {
+          this.bootstrapper.update(cache)
```

Comment:
The subscription is now here only to update the bootstrapper?

Reply:
Correct

Reply:
I wonder if it makes sense to pass it to `patch-db` to be added here:
```ts
  public cache$ = this.source$.pipe(
    tap(res => this.store.update(res)),
    map(_ => this.store.cache),
    tap(cache => this.bootstrapper.update(cache),
    shareReplay(1),
  )
```
Because we're not really *starting* anything here...

Reply:
I'm game, will require some thought as to keeping PatchDB generic with regard to initial cache and bootstrapper updating.